src/student/student.service.ts

```diff
@@ -3,21 +3,96 @@ import { CreateStudentDto } from './dto/create-student.dto';
 import { UpdateStudentDto } from './dto/update-student.dto';
 import { ReadStudentDto } from './dto/read-student.dto';
 import { PrismaService } from 'src/prisma/prisma.service';
+import { createClient } from '@supabase/supabase-js';
+import { decode } from 'base64-arraybuffer';
+import { v4 as uuidv4 } from 'uuid';
 
 @Injectable()
 export class StudentService {
   constructor(private prisma: PrismaService) {}
 
-  create(createStudentDto: CreateStudentDto) {
-    const testStudent = this.prisma.testModel.create({
+  getPaymentById(createStudentDto) {
+    return this.prisma.payment.findUnique({
+      where: {
+        id: createStudentDto.paymentId,
+      },
+    });
+  }
+
+  async createPayment(payment_path: string, isRegisterByStudent: boolean) {
+    const newPayment = await this.prisma.payment.create({
+      data: {
+        photo_src: payment_path,
+        // if submission is by student, put in pending status
+        // until cashier accepts payment
+        // else submission is by cashier, accepted status
+        status: isRegisterByStudent ? 'pending' : 'accepted',
+      },
+    });
+    return newPayment;
+  }
+
+  async createEvent() {
+    const newEvent = await this.prisma.event.create({
+      data: {
+        title: '',
+        price: '',
+        max_participants: 1,
+        requires_payment: true,
+        description: '',
+        date: new Date(),
+        is_active: true,
+        form_name: '',
+      },
+    });
+    return newEvent;
+  }
+
+  async createStudent(
+    createStudentDto: CreateStudentDto,
+    file: Express.Multer.File,
+  ) {
+    const uuid = uuidv4();
+    const payment_path = this.uploadImageToDB(file, uuid);
+    createStudentDto.uuid = uuid;
+    const isStudent = true;
+    const payment = await this.createPayment(await payment_path, isStudent);
+    const event = await this.createEvent();
+    createStudentDto.paymentId = payment.id;
+    createStudentDto.eventId = event.id;
+    const newStudent = this.prisma.student.create({
       data: createStudentDto,
     });
+    return newStudent;
+  }
 
-    return testStudent;
+  toBase64(file: Express.Multer.File) {
+    return Buffer.from(file.buffer).toString('base64');
+  }
+
+  async uploadImageToDB(file: Express.Multer.File, uuid: string) {
```

Comment:
this function should be inside supabaseservice also pa edit ng .env.example sa mga ginamit mo na env variables

Reply:
Questions:
1. Asa na folder nako i create ang supabase service na file? Sa may supabase na folder? or sa student lang pud?
2. Pwede ko pa explain further unsa pasabot nimo sa pa edit sa .env? I change nako ang `process.env.SUPABASE_SERVICE_ROLE_KEY, process.env.SUPABASE_URL`? 
@rhaeyx

Reply:
1. Sa 1-login-xxx na branch kay nag gamit siya supabaseService didto, pwede ka mag refer ato pero modify lang as you see fit.

2. Pa set lang ko ug placeholders sa .env.example na file, like add tung mga variables na gidungag nimo sa .env
```
...
DATABASE_URL="postgresql://postgres:postgres@localhost:54322/postgres?schema=public"
DIRECT_URL="postgresql://postgres:postgres@localhost:54322/postgres?schema=public"
SUPABASE_URL="http://127.0.0.1:54321"
SUPABASE_ANON_KEY="supabase_anon_key"
SUPABASE_SERVICE_ROLE_KEY="supabase_service_role_key"
```
so ma something like this na ang .env.example

Reply:
Resolved, thanks!
Action:
-Moved uploadImageToDB() and toBase64() functions to supabase.service
-Copied and created supabase.service and supabase.module from issue 1 (might merge conflict)
-Added new placeholders to .env.example file